source/robot_model/include/robot_model/Model.hpp

```diff
@@ -157,30 +179,22 @@ class Model {
   const StateRepresentation::CartesianTwist forward_kinematic(const StateRepresentation::JointState& joint_state);
 
   /**
-   * @brief Compute the inverse kinematic, i.e. joint velocities from the velocities of the frames in parameter
+   * @brief Compute the inverse kinematics, i.e. joint velocities from the velocities of the frames in parameter
    * @param joint_state usually the current joint state, used to compute the jacobian matrix
    * @param cartesian_states vector of twist
    * @return the joint velocities of the robot
    */
-  const StateRepresentation::JointVelocities inverse_kinematic(const StateRepresentation::JointState& joint_state,
-                                                               const std::vector<StateRepresentation::CartesianState>& cartesian_states);
+  StateRepresentation::JointVelocities inverse_kinematic(const StateRepresentation::JointState& joint_state,
+                                                         const std::vector<StateRepresentation::CartesianState>& cartesian_states);
 
   /**
    * @brief Compute the inverse kinematic, i.e. joint velocities from the twist of the end-effector
    * @param joint_state usually the current joint state, used to compute the jacobian matrix
    * @param cartesian_state containing the twist of the end-effector
    * @return the joint velocities of the robot
    */
-  const StateRepresentation::JointVelocities inverse_kinematic(const StateRepresentation::JointState& joint_state,
-                                                               const StateRepresentation::CartesianState& cartesian_state);
-
-  /**
-   * @brief Compute the inverse kinematic, i.e. joint velocities from the twist of the end-effector as Eigen vectors
-   * @param joint_positions current joint positions to compute the jacobian
-   * @param cartesian_twist the twist of the end-effector
-   * @return the joint velocities of the robot
-   */
-  const Eigen::VectorXd inverse_kinematic(const Eigen::VectorXd& joint_positions, const Eigen::VectorXd& cartesian_twist);
```

Comment:
This function should be discarded in my opinion. Either we work with state_representation or not. For consistency.

Reply:
Agreed, we can leave the conversion to Eigen types to the StateRepresentation level. For example, it returns StateRepresentation::JointVelocities which has an inbuilt cast / conversion method to Eigen.

Reply:
agreed as well